### PR TITLE
Sons of init bug

### PR DESCRIPTION
**Pull request by nastasi-oq**
Add a fix to manage forked processes when parent process crashes or is killed.
Add infrastructure to run bash tests and retrieve xunit reports.
Tests for 'precise' are green: https://ci.openquake.org/job/zdevel_oq-risklib/1325/
Tests for 'trusty' are green: https://ci.openquake.org/job/zdevel_oq-risklib/1326/

### CODE REVIEW

File: openquake/commonlib/concurrent_futures_process_mpatch.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+                e = sys.exc_info()[1]
+                result_queue.put(_ResultItem(call_item.work_id,
+                                             exception=e))
+            else:
+                result_queue.put(_ResultItem(call_item.work_id,
+                                             result=r))
+
+    def concurrent_futures_process_monkeypatch():
+        global process, _process_worker
+
```
Comment (micheles) [inline openquake/commonlib/concurrent_futures_process_mpatch.py]: There is no need to declare the globals

File: openquake/risklib/tests/bin/bash_tests.sh [hunk 2]
```diff
@@ -0,0 +1,117 @@
+#!/bin/bash
+# set -x
+declare -a available_tests
```
Comment (micheles) [inline openquake/risklib/tests/bin/bash_tests.sh]: Maybe add a comment explaining the logic of this script and how dumb.sh is used

File: openquake/risklib/tests/bin/bash_tests.sh [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        fi
+    fi
+done
+if [ $some_test -eq 0 ]; then
+    echo "No test started"
+    usage 2
+fi
+if [ "$outfile" ]; then
+    xunit_output >$outfile
+fi
```
Comment (daniviga) [inline openquake/risklib/tests/bin/bash_tests.sh]: New line missing

File: openquake/risklib/tests/bin/dumb.sh [hunk 4]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+dumb()
+{
+    echo "Dumb is here"
+    return 0
+}
```
Comment (daniviga) [inline openquake/risklib/tests/bin/dumb.sh]: New line missing

File: openquake/risklib/tests/bin/sons-of-init.sh [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+# # cleanup
+if [ $survived_number -ne 0 ]; then
+    IFS='
+'
+    for i in $(echo "$survived" | grep -v '^$'); do
+        kill -KILL $i
+    done
+fi
+
+exit $survived_number
```
Comment (daniviga) [inline openquake/risklib/tests/bin/sons-of-init.sh]: New line missing

File: debian/changelog [hunk 6]
```diff
@@ -1,3 +1,9 @@
+  [Matteo Nastasi]
+  * Add framework to run bash scripts tests and collect associated xunit report
+  * Add concurrent_futures_process_monkeypatch() using prctl library to avoid
```
Comment (daniviga) [inline debian/changelog]: prctl -> python-prctl sice those are two different libraries

**Comment by daniviga** [comment]
A part from minor things, LGTM

File: openquake/risklib/tests/bin/bash_tests.sh (hunk 3 above)
Comment (nastasi-oq) [inline openquake/risklib/tests/bin/bash_tests.sh]: Done

File: openquake/risklib/tests/bin/dumb.sh (hunk 4 above)
Comment (nastasi-oq) [inline openquake/risklib/tests/bin/dumb.sh]: Done

File: openquake/risklib/tests/bin/sons-of-init.sh (hunk 5 above)
Comment (nastasi-oq) [inline openquake/risklib/tests/bin/sons-of-init.sh]: Done

**Comment by micheles** [comment]
LGTM

**Pull request merged**